### ISSUE TITLE
Remove PartSpec and use reflection

### ISSUE THREAD

**Issue by db47h** [db47h/hwsim#10]
For custom components, what if we did something like this:

```go
type Mux8 struct {
    In  [8]int `hw:in`
    Sel int    `hw:in`
    Out [8]int `hw:out`
}

func (m *Mux8) Update(c *Circuit) {
    // ...
}

// Along with a Component interface
type Component interface {
    Update(c *Circuit)
}
```

And use `reflect` to get the specs of a part. `Chip.mount` could then instantiate parts as needed and init their input/outputs.

**Comment by db47h** [comment]
Working prototype: https://gist.github.com/db47h/de370b53213766c4890a3f7afcb83253

**Comment by db47h** [comment]
The prototype works _with_ `PartSpec`. Plus it's les flexible than `PartSpec`:

Cons:
- Flexibility: How to implement the simple gates without a lot of boiler plate code?

Pros:
- No need to implement `PartSpec.Mount`
- Cleaner

**Comment by db47h** [comment]
Merged 34622acbce997e5eb56f24a9d287f15754279089
